tests/unit_tests/pyanaconda_tests/modules/storage/partitioning/test_module_part_automatic.py

```diff
@@ -283,6 +288,106 @@ def test_get_partitioning(self, platform, suggest_swap_size):
         assert [Size("1GiB"), Size("1GiB"), Size("500MiB")] == \
             [spec.size for spec in requests]
 
+    @patch('pyanaconda.modules.storage.partitioning.automatic.utils.platform')
+    @patch('pyanaconda.modules.storage.partitioning.automatic.utils.conf')
+    @patch('pyanaconda.modules.storage.partitioning.automatic.automatic_partitioning.suggest_swap_size')
+    def test_get_partitioning_hibernation(self, suggest_swap_size, mocked_config, platform):
+        """Test the creation of swap with PartitioningRequest.hibernation."""
+        suggest_swap_size.return_value = Size("1GiB")
+
+        storage = create_storage()
+        platform.partitions = [
+            PartSpec(mountpoint="/boot", size=Size("1GiB"))
+        ]
+
+        # Test Case: No swap, hibernation
+        mocked_config.storage.default_partitioning = [
+            {
+                'name': '/',
+                'size': Size("50 GiB"),
+            },
+        ]
+        partitioning_request = PartitioningRequest()
+        partitioning_request.hibernation = True
+
+        requests = AutomaticPartitioningTask._get_partitioning(
+            storage=storage,
+            scheme=AUTOPART_TYPE_LVM,
+            request=partitioning_request
+        )
+
+        swap = None
+        for spec in requests:
+            if spec.fstype == "swap":
+                swap = spec
+                break
+
+        assert swap is not None
```

Comment:
You can remove the loop and check `assert any(spec for spec in requests if spec.fstype == "swap")`. Same below. It means that there has to be at least one spec with the swap file system type.

Reply:
Thanks, this is much better.